test_data/swc/custom_type.swc

```diff
@@ -0,0 +1,11 @@
+# Use custom type: 7
```

Comment:
what does this do to iteration when custom types exist? what about when undefined?

Reply:
What do you mean ?

Reply:
What would the user expect when iterating over NeuriteType.all?  Same with the other helpers, ie: is 'custom' a dendrite?

Reply:
That's a tough question. Two answers:
_ Yes, custom is a dendrite. There are little chances people are gonna define custom axon types. And if we don't do this all morphometrics for morphologies with custom types are gonna be broken.
_ No, custom is not a dendrite. all should only refer to properly defined types apical_dendrite and basal_dendrite

or should we rename custom to dendrite_custom to make it more obvious ?

Reply:
Is it possible to allow the user define a type mapping?

For example start with a default one, where anything else apart from the defined types is undefined unless the user passes their own types. BUT these types correspond to the types found in the file not an arbitrary set of integers.
```
class DefaultTypes(object):
    undefined = 0
    soma = 1
    axon = 2
    basal_dendrite = 3
    apical_denrite = 4
```
or the user could provide:
```
class UserTypes(object):
    undefined = 0
    soma = 1
    endfoot = 2
    glia_process = 3
    exotic_process = 7
```

Reply:
The more we discuss about this the more I think we need different loader function for each type of cell.
Instead of allowing user define type mapping having AstrocyteSectionType that would be set in stones. And a function load_astrocyte that would be the equivalent of load_neuron but that would return sections of type AstrocyteSectionType ?

Reply:
I just talked to @eleftherioszisis and the use case he's running into is that some of the reconstructions are mis-numbered; ie: 7 is used for `endfoot` instead of 2 (?).  Instead of having a unique loader, one option, I think, would be to pass a mapping into a loader that says what types to map to the internal types:

```
import neurom as nm
nrn = nm.load_neuron('some/data/path/morph_file0.swc', mapping={7: NeuriteType.axon, ...})
```

This would partly solve his problem, however, it doesn't solve the problem of having to use the NeuroM typenames (axon/dendrite/etc) for Glia concepts (endfoot/process/etc), but it would allow for more flexibility when loading files.

Reply:
That proposal is consistent with the tree_type definition that happens during loading in TMD morphologies. It would be useful if this is possible to extend to MorphIO so all approaches are synchronized.

Reply:
Yeah, that will provide us with much more flexibility in cases of custom typing. I am in favor also in favor of @wizmer suggestion for a different loader for astrocytes. However, this will imply that features will have to be further classified by `cell_type`.

Reply:
> I just talked to @eleftherioszisis and the use case he's running into is that some of the reconstructions are mis-numbered; ie: 7 is used for `endfoot` instead of 2 (?). Instead of having a unique loader, one option, I think, would be to pass a mapping into a loader that says what types to map to the internal types:
> 
> ```
> import neurom as nm
> nrn = nm.load_neuron('some/data/path/morph_file0.swc', mapping={7: NeuriteType.axon, ...})
> ```
> 
> This would partly solve his problem, however, it doesn't solve the problem of having to use the NeuroM typenames (axon/dendrite/etc) for Glia concepts (endfoot/process/etc), but it would allow for more flexibility when loading files.

From what you are saying, the morphology is broken, I don't think it is NeuroM's role to repair it. I'd rather provide a CLI tool neurite-type-fixer to fix broken morphologies than open a Pandora box by allowing people to pass custom type mappings.

Reply:
> However, this will imply that features will have to be further classified by `cell_type`.
Yes, and that actually makes sense